src/commands/staking/delegate/unstake.rs

```diff
@@ -0,0 +1,88 @@
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = super::DelegateStakeContext)]
+#[interactive_clap(output_context = UnstakeContext)]
+pub struct Unstake {
+    /// Enter the amount to unstake from the inner account of the predecessor (example: 10NEAR or 0.5near or 10000yoctonear):
+    amount: crate::common::NearBalance,
+    #[interactive_clap(skip_default_input_arg)]
+    /// What is the signer account ID?
+    signer_account_id: crate::types::account_id::AccountId,
+    #[interactive_clap(named_arg)]
+    /// Select network
+    network_config: crate::network_for_transaction::NetworkForTransactionArgs,
+}
+
+#[derive(Clone)]
+pub struct UnstakeContext(crate::commands::ActionContext);
+
+impl UnstakeContext {
+    pub fn from_previous_context(
+        previous_context: super::DelegateStakeContext,
+        scope: &<Unstake as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let signer = scope.signer_account_id.clone();
+        let signer_id: near_primitives::types::AccountId = scope.signer_account_id.clone().into();
+        let validator_account_id = previous_context.validator_account_id.clone();
+        let interacting_with_account_ids = vec![validator_account_id.clone()];
+        let amount = scope.amount.clone();
+
+        let on_after_getting_network_callback: crate::commands::OnAfterGettingNetworkCallback =
+            std::sync::Arc::new(move |_network_config| {
+                Ok(crate::commands::PrepopulatedTransaction {
+                    signer_id: signer_id.clone(),
+                    receiver_id: previous_context.validator_account_id.clone(),
+                    actions: vec![near_primitives::transaction::Action::FunctionCall(
+                        near_primitives::transaction::FunctionCallAction {
+                            method_name: "unstake".to_string(),
+                            args: serde_json::json!({
+                                "amount": amount.clone().to_yoctonear().to_string()
+                            })
+                            .to_string()
+                            .into_bytes(),
+                            gas: crate::common::NearGas::from_tgas(300).as_gas(),
+                            deposit: 0,
+                        },
+                    )],
+                })
+            });
+
+        let amount = scope.amount.clone();
+        let on_after_sending_transaction_callback: crate::transaction_signature_options::OnAfterSendingTransactionCallback = std::sync::Arc::new(
+                move |outcome_view, _network_config| {
+                    if let near_primitives::views::FinalExecutionStatus::SuccessValue(_) = outcome_view.status {
+                        eprintln!("<{signer}> has successfully unstake {amount} from <{validator_account_id}>.")
+                    }
+                    Ok(())
+                },
+            );
```

Comment:
Let's move locally cloned variables into inner block:

```suggestion
        let on_after_sending_transaction_callback: crate::transaction_signature_options::OnAfterSendingTransactionCallback = std::sync::Arc::new({
                let amount = scope.amount.clone();

                move |outcome_view, _network_config| {
                    if let near_primitives::views::FinalExecutionStatus::SuccessValue(_) = outcome_view.status {
                        eprintln!("<{signer}> has successfully unstake {amount} from <{validator_account_id}>.")
                    }
                    Ok(())
                }
            });
```

Note: It is a bit hard to read it here, but in the nutshell, it is just one extra logical block of `{}`, where the last expression is the return value of the block:

```rs
let x = {
    let amount = scope.amount.clone();
    amount + 1
};
```